Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithm.H

```diff
@@ -66,12 +69,19 @@ class PsatdAlgorithm : public SpectralBaseAlgorithm
          */
         virtual int getRequiredNumberOfFields () const override final
         {
-            if (m_time_averaging) {
-                return SpectralAvgFieldIndex::n_fields;
-            } else {
+            if (!m_J_linear_in_time && !m_time_averaging)
```

Comment:
Most people find concatinated negations a bit hard to read, this equivalent logic can be a bit simpler:
```suggestion
            if (m_J_linear_in_time || m_time_averaging)
```

Reply:
I have tried to restructure that in https://github.com/ECP-WarpX/WarpX/pull/1828/commits/716f907bbdbb4f921130bf56e2244b34c3f229d4. The first old `if` block was accessed only when both `m_J_linear_in_time = false` and `m_time_averaging = false`, not when either one of the two was `true`. I think we would have needed a `!` in front of everything, if we wanted to use `||`. Instead of doing that, I rewrote the whole thing upside-down, so that the `if` always applies only to one variable:
```cpp
if (m_J_linear_in_time)
{
    return SpectralFieldIndexJLinearInTime::n_fields;
}
else
{
    if (m_time_averaging)
    {
        return SpectralFieldIndexTimeAveraging::n_fields;
    }
    else
    {
        return SpectralFieldIndex::n_fields;
    }
}
```
This should be correct and I think it is clearer. Let's see if CI catches any issues. Note that we might be able to get rid of this logic in some of the follow-up PRs.